### PR TITLE
Update blitz-next docs

### PR DESCRIPTION
**Pull request by dillonraphael**

### CODE REVIEW

File: app/pages/docs/blitzjs-next.mdx [hunk 1]
```diff
@@ -136,8 +148,24 @@ setupBlitzServer({
 
 An object with the [`gSSP`](#blitzjs-next-gssp), [`gSP`](#blitzjs-next-gsp) & [`api`](#blitzjs-next-api) wrappers.
 
+### Custom Server {#custom-next-server}
+
+The Blitz CLI supports running custom Next.js servers. This means you can compile typescript and use the blitz cli to inject env variables. By default, the CLI checks for `server/index.ts` or `server.ts`
```
Comment (beerose) [inline app/pages/docs/blitzjs-next.mdx:153]: Why not javascript? You only mention TypeScript. Also, you should link to Next.js docs

File: app/pages/docs/blitzjs-next.mdx [hunk 2]
```diff
@@ -184,3 +212,146 @@ export default api(async (req, res, ctx) => {
...
+
+
+## Concepts {#blitzjs-next-concepts}
+
+### Authenticate user before page loads {#authenticate-users-on-page-load}
+
+You may want to check if the user is logged in before your page loads. We’re going to use the `getCurrentUser` query inside `getServerSideProps()` by calling the query directly. Then you can check if the user is logged in on the server and use the built-in Next.js redirect property.
+
+```ts
+...
```
Comment (beerose) [inline app/pages/docs/blitzjs-next.mdx:224]: What are these dots?

File: app/pages/docs/blitzjs-next.mdx [hunk 3]
```diff
@@ -184,3 +212,146 @@ export default api(async (req, res, ctx) => {
...
+      }
+    }
+  }
+
+})
+...
+```
+
+### Return types when data fetching on the server {#data-fetching-types}
+You can set the types returned from the Next.js data fetching functions. All Blitzjs wrappers for the Nextjs functions accept a generic. Same with the `BlitzPage` type.
```
Comment (beerose) [inline app/pages/docs/blitzjs-next.mdx:252]:
```suggestion
You can set the types returned from the Next.js data fetching functions. All Blitz.js wrappers for the Next.js functions accept a generic. Same with the `BlitzPage` type.
```

File: app/pages/docs/blitzjs-next.mdx [hunk 4]
```diff
@@ -184,3 +212,146 @@ export default api(async (req, res, ctx) => {
...
+        <p>User Page</p>
+        {user && <p>{user.email}</p>}
+      </div>
+    </Layout>
+  )
+}
+
+export default Page
+
+...
```
Comment (beerose) [inline app/pages/docs/blitzjs-next.mdx:297]: ??

File: app/pages/docs/blitzjs-next.mdx [hunk 5]
```diff
@@ -184,3 +212,146 @@ export default api(async (req, res, ctx) => {
...
+```
+
+### Return types when data fetching on the server {#data-fetching-types}
+You can set the types returned from the Next.js data fetching functions. All Blitzjs wrappers for the Nextjs functions accept a generic. Same with the `BlitzPage` type.
+
+So for example, we can use some typescript utilities to help use get the types returned by `getCurrentUser()`
+
+```ts
+...
```
[the remark this answers is not in the document]
Comment (dillonraphael) [reply app/pages/docs/blitzjs-next.mdx:257]: The dots are abbreviation. Should I remove?

**Review by beerose — CHANGES_REQUESTED** [review]
Please run the linter on the changed file!

**Review by beerose — APPROVED** [review]

**Pull request merged**